flexget/plugins/input/my_anime_list.py

```diff
@@ -58,67 +60,43 @@ class MyAnimeList(object):
+        selected_types = config.get('type', list(ANIME_TYPE))
 
         if not isinstance(selected_status, list):
             selected_status = [selected_status]
 
         if not isinstance(selected_types, list):
             selected_types = [selected_types]
 
+        for i in range(len(selected_status)):
+            selected_status[i] = STATUS[selected_status[i]]
+
         try:
-            list_response = task.requests.get('https://myanimelist.net/malappinfo.php', params=parameters)
+            list.response = task.requests.get('https://myanimelist.net/animelist/' + config['username'] + '/load.json')
         except RequestException as e:
             raise plugin.PluginError('Error finding list on url: {url}'.format(url=e.request.url))
 
         try:
-            tree = ET.fromstring(list_response.text.encode('utf-8'))
-            list_items = tree.findall('anime')
-        except ET.ParseError:
-            raise plugin.PluginError('Bad XML')
-
-        for item in list_items:
-            my_anime_list_id = item.findtext('series_animedb_id')
-            title = item.findtext('series_title').strip()
-            anime_type = ANIME_TYPE[item.findtext('series_type', 1)]
-            my_status = STATUS[item.findtext('my_status')]
-
-            my_tags = []
-            alternate_names = []
-            is_exact = False
-
-            for name in item.findtext('series_synonyms', '').split('; '):
-                stripped = name.strip()
-                if stripped and stripped is not title:
-                    alternate_names.append(name)
-
-            for tag in item.findtext('my_tags', '').split(','):
-                stripped = tag.strip()
-                if stripped:
-                    my_tags.append(stripped)
-                    if stripped is 'exact':
-                        is_exact = True
-
-            # if user has chosen a status or a type, match strictly, otherwise let it all through
-            wanted_status = not selected_status or my_status in selected_status
-            wanted_type = not selected_types or anime_type in selected_types
-
-            if wanted_status and wanted_type:
-                entry = Entry(title=title,
-                              url='https://myanimelist.net/anime/{}'.format(my_anime_list_id),
-                              configure_series_alternate_name=alternate_names,
-                              my_anime_list_type=anime_type,
-                              my_anime_list_status=my_status,
-                              my_anime_list_tags=my_tags)
-
-                if is_exact:
-                    entry['configure_series_exact'] = True
-
-                if entry.isvalid():
-                    entries.append(entry)
-
+            js = json.loads(list.response.text.encode('utf-8'))
+        except:
+            raise plugin.PluginError('Error reading JSON')
+
+        for anime in js:
+            if (anime["status"] in selected_status
```

Comment:
Funky indentation

Reply:
Broke it down to comply with [styling recommendation](https://www.python.org/dev/peps/pep-0008/#indentation) (up to 120 characters line) and make the conditions easier to read